app/controllers/application_controller.rb

```diff
@@ -5,6 +5,157 @@ class ApplicationController < Sinatra::Base
   configure do
     set :public_folder, 'public'
     set :views, 'app/views'
+    enable :sessions
+    set :session_secret, "fwitter"
+  end
+  
+  get '/' do
+    erb :index
+  end
+  
```

Comment:
Consider creating helper methods in your ApplicationController to grab the current user and to determine if a user is logged in. 

```
helpers do
    def logged_in?
      !!session[:user_id]
    end

    def current_user
      User.find(session[:user_id])
    end

  end
```